Sources/RTMP/RTMPConnection.swift

```diff
@@ -455,12 +457,26 @@ open class RTMPConnection: EventDispatcher {
         previousTotalBytesIn = totalBytesIn
         previousTotalBytesOut = totalBytesOut
         previousQueueBytesOut.append(socket.queueBytesOut.value)
+        
+        
+        if currentBytesOutPerSecond == 0 {
+            zeroBytesOutCounter += 1
+            // If currentBytesOutPerSecond has been 0 for the last 5 seconds
+            if zeroBytesOutCounter >= 5 {
+                logger.warn("STALE connection, force close")
+                self.close(isDisconnected: true) // call close function
+                zeroBytesOutCounter = 0 // reset counter
+            }
+        } else {
+            zeroBytesOutCounter = 0 // reset counter
+        }
```

Comment:
During playback, there are cases where there is no data transfer for more than 5 seconds, which results in strong side effects. Due to this, it's not acceptable.

I would like this to be handled within the delegate method.
https://github.com/shogo4405/HaishinKit.swift/blob/main/Sources/RTMP/RTMPConnection.swift#L478

Reply:
I'm considering whether these parameters around here can be used for handling this situation.
https://developer.apple.com/documentation/network/nwprotocoltcp/options#3225423

Reply:
OK, sorry didn't consider playback. 
As for using NWProtocolTCP options,  keep in mind, they're being overwritten here:  https://github.com/shogo4405/HaishinKit.swift/blob/3db54f064f3d66c18ffa3163dd771a5aec2cff63/Sources/RTMP/RTMPNWSocket.swift#L25 so it's impossible to set parameters from the outside.
I've tried setting them inside but it didn't affect the hanging back then. But can actually work now. Imho these timeouts should be very short, about 5s

Reply:
I investigated, and it seems that the value of options.connectionDropTime is related to the switch from WiFi to Cellular. However, in any case, viabilityDidChange(to:) is always triggered, and I have confirmed that handling this method allows for proper closure without specifically checking connectionDropTime.

```diff
diff --git a/Sources/RTMP/RTMPConnection.swift b/Sources/RTMP/RTMPConnection.swift
index f89bc609..e9132df5 100644
--- a/Sources/RTMP/RTMPConnection.swift
+++ b/Sources/RTMP/RTMPConnection.swift
@@ -458,6 +458,7 @@ open class RTMPConnection: EventDispatcher {
         for stream in streams {
             stream.on(timer: timer)
         }
+        print(currentBytesOutPerSecond)
         if measureInterval <= previousQueueBytesOut.count {
             var total = 0
             for i in 0..<previousQueueBytesOut.count - 1 where previousQueueBytesOut[i] < previousQueueBytesOut[i + 1] {
diff --git a/Sources/RTMP/RTMPNWSocket.swift b/Sources/RTMP/RTMPNWSocket.swift
index 9155b329..14c31185 100644
--- a/Sources/RTMP/RTMPNWSocket.swift
+++ b/Sources/RTMP/RTMPNWSocket.swift
@@ -74,6 +74,27 @@ final class RTMPNWSocket: RTMPSocketCompatible {
         totalBytesOut.mutate { $0 = 0 }
         queueBytesOut.mutate { $0 = 0 }
         inputBuffer.removeAll(keepingCapacity: false)
+
+        let options = NWProtocolTCP.Options()
+        print(options.enableKeepalive)
+        options.connectionTimeout = 5
+        options.connectionDropTime = 10
+        options.persistTimeout = 15
+
+        print(parameters.debugDescription)
+
+        let tls = NWProtocolTLS.Options()
+        sec_protocol_options_set_verify_block(tls.securityProtocolOptions, { _, sec_trust, sec_protocol_verify_complete in
+            let trust = sec_trust_copy_ref(sec_trust).takeRetainedValue()
+            var error: CFError?
+            if SecTrustEvaluateWithError(trust, &error) {
+                sec_protocol_verify_complete(true)
+            } else {
+                sec_protocol_verify_complete(true)
+            }
+        }, self.networkQueue)
+
+        parameters = NWParameters(tls: tls, tcp: options)
         connection = NWConnection(to: NWEndpoint.hostPort(host: .init(withName), port: .init(integerLiteral: NWEndpoint.Port.IntegerLiteralType(port))), using: parameters)
         connection?.viabilityUpdateHandler = viabilityDidChange(to:)
         connection?.stateUpdateHandler = stateDidChange(to:)
```